src/app/guards/auth.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { UserService } from '@app/services/user.service';
+
+@Injectable({
+  providedIn: 'root'
```

Comment:
ok, you are providing twice because in this line you say inject on root, but on app,module.ts you are providing the service again